posts/GeoMx-scMask-generation/index.qmd

```diff
@@ -0,0 +1,167 @@
+---
+title: "GeoMx&#8482; mask generation for marker-based single-cell application"
+author:
+  - name: Lidan Wu
+    orcid: 0000-0003-3150-6170
+    affiliations:
+      - ref: nstg
+      - ref: lidanwu
+toc: true
+toc-title: Contents
+toc-depth: 3
+toc-expand: 2
+toc-location: left
+number-sections: true
+number-depth: 4
+date: "2024-06-12"
+date-modified: last-modified
+categories: [GeoMx, how-tos, image processing, Python]
+image: figures/workflow_GeoMx_scMask_given_query.png
+description: "This post introduces a pipeline for automatic generating GeoMx&#8482;-ready binary masks in batch for marker-based single-cell application. Given the query marker protein of interest, the pipeline would take morphology images and generate binary masks for negative-stained cells and cells connecting to positive-stained regions, respectively. The pipeline runs as a command line and this post would serve as a guide to how it works and how to use it."
+draft: false
+execute:
+  engine: knitr
+---
+
+# Introduction
+
+Ability to separate cells based on their relationship to nearby marker protein staining is a desired feature in many biological studies. On such use case would be to study the difference between neurons immediately next to neurofibrillary tangles and neurons that are a little further away. This post introduces a pipeline for automatic generating GeoMx&#8482;-ready binary masks in batch for marker-based single-cell application.
+
+Like other items in our [Analysis Scratch Space](https://nanostring-biostats.github.io/CosMx-Analysis-Scratch-Space/about.html){target="_blank"},
+the usual [caveats](https://nanostring-biostats.github.io/CosMx-Analysis-Scratch-Space/about.html){target="_blank"} and [license](https://nanostring-biostats.github.io/CosMx-Analysis-Scratch-Space/license.html){target="_blank"} applies.
+
+
+
+# Workflow Description
+
+The pipeline is a python script `Generate_scMask_given_query.py` that runs as a command line and you can downloaded from `_code/GeoMx-scMask-generation` [folder](https://github.com/Nanostring-Biostats/CosMx-Analysis-Scratch-Space/tree/Main/_code/GeoMx-scMask-generation){target="_blank"} in this repository. By passing file path to image folder to the script, the pipeline would take GeoMx&#8482; morphology images and generate binary masks for negative-stained cells and cells connecting to positive-stained regions, respectively. 
+
+![Workflow schematic](figures/workflow_GeoMx_scMask_given_query.png){#fig-workflow}
```

Comment:
Can you be more descriptive here so that someone with a neuro background can follow. For example, what are Cyto and Query staining. I'm having trouble relating this figure to the initial example in the introduction.

Reply:
I added a more descriptive caption for the figure and now it reads:

> Workflow schematic. In neuropathology study, it's sometimes desired to separate between neurons (stained by NeuN in red channel) touching the neurofibrillary tangles (stained by Phospho-Tau in green channel) and those that are a little further away. The pipeline described here fullfills the need of separating cells based on additional morphology marker of interest (aka "Query") and could also be applied to other single-cell applications like separating immune cells based on their contact with nearby pathogen.